internal/config/config.go

```diff
@@ -869,7 +875,26 @@ func getConfigData(namespace string, env Env, blockchain common.Blockchain, netw
 		return reader, nil
 	}
 
-	configPath := fmt.Sprintf("%v/%v/%v/%v.yml", namespace, blockchainName, networkName, env)
+	configPath := GetConfigPath()
+	// If configPath is not set, try to construct the file system path from configRoot.
+	if len(configPath) == 0 && len(configRoot) > 0 {
+		configPath = fmt.Sprintf("%v/%v/%v/%v/%v.yml", configRoot, namespace, blockchainName, networkName, env)
+		if sidechain != api.SideChain_SIDECHAIN_NONE {
+			configPath = fmt.Sprintf("%v/%v/%v/%v/%v/%v.yml", configRoot, namespace, blockchainName, networkName, sidechainName, env)
+		}
+	}
+
+	// If either configRoot or configPath is set, read the config from the file system.
+	if len(configPath) > 0 {
```

Comment:
if we use configRoot, then the config folder will have to have the ethereum/mainnet/base.yml structure.  Letting user specify the configPath to the config file they want to load seem a lot more straight forward and flexible than prescribing the config folder hierarchy.  It's going to be easier to configure and figure the config used for a particular machine as well.

Reply:
It's a simple trade off, having both option, the users have options to choose how they want to structure their configuration.  The added complexity is really minimal by having configPath.

Maybe it's minor point, but having a single config path, there's only 1 file to look and the config file would be at a fix location.  It can be easier to setup and easier to read the config instead of reading multiple files.

Is there some significant downside to allowing configPath?